src/status_im/subs/wallet/wallet.cljs

```diff
@@ -261,22 +263,22 @@
  :wallet/settings-keypairs-accounts
  :<- [:wallet/keypairs]
  (fn [keypairs [_ format-options]]
-   (let [grouped-keypairs     (group-by :lowest-operability keypairs)
+   (let [grouped-keypairs     (->> keypairs
+                                   vals
+                                   (group-by :lowest-operability))
          operable-keypair-ids (->> (concat (:fully grouped-keypairs)
                                            (:partially grouped-keypairs))
-                                   (map :key-uid)
-                                   (into #{}))
-         missing-keypair-ids  (->> (map :key-uid (:no grouped-keypairs))
-                                   (into #{}))]
-     {:operable (->> keypairs
-                     (filter #(contains? operable-keypair-ids (:key-uid %)))
+                                   (mapv :key-uid))
+         missing-keypair-ids  (mapv :key-uid (:no grouped-keypairs))]
+     {:operable (->> (select-keys keypairs operable-keypair-ids)
+                     vals
                      (map (fn [{:keys [accounts name type key-uid]}]
                             {:type     (keyword type)
                              :name     name
                              :key-uid  key-uid
                              :accounts (format-settings-keypair-accounts accounts format-options)})))
-      :missing  (->> keypairs
-                     (filter #(contains? missing-keypair-ids (:key-uid %)))
+      :missing  (->> (select-keys keypairs missing-keypair-ids)
+                     vals
                      (map (fn [{:keys [accounts name type key-uid]}]
                             {:type     (keyword type)
                              :name     name
```

Comment:
This event is performing operations that are very slow in Clojure, such as `group-by`, `select-keys`, `concat` + `mapv`.
It'd be great if we measure if this event is taking too much to compute.

To reduce the transformations in this last piece of code that is using `select-keys` + `vals` + `map`, Consider that we could use a single  call to `keep`:
```clojure
(let [missing-keypair-ids (set 'something)]
  (keep (fn [[keypair-id {,,,
                          key-uid-name :name
                          key-uid-type :type}]]
          (when (missing-keypair-ids keypair-id)
            {:type (keyword key-uid-type)
             :name key-uid-name
             ,,,}))
        {}))
```

Reply:
We may need to check for multiple keypairs.

> I need to run the sub 1000 times to start to get into slow territory (25ms). Not bad.

So, is a single run pretty fast? I don't get it, but if a single run is above 16ms it's not good.

Reply:
No no, 1000 runs ~25ms, thus one run of this subscription with few keypairs is almost nothing. The absolute time can never be trusted, but coming from emulated code, one run of this sub should very likely take less than a millisecond to process, unless we're going to process significantly larger payloads here.

You guys can decide better. If it's an obvious performance refactor I think it's worth it (like replacing `select-keys` with a hardcoded set of keys when doable). But I would be more cautious of refactoring more aggressively if the code is already fast, e.g. packing everything into a single reduce pass to iterate once and avoid spurious group-by and so on. This kind of optimization usually only pays off when there are more items in the collections being processed.

From the perspective of orders of magnitude, ff a collection only has ~10 elements in it, it's probably nothing to worry in CLJS subscriptions. 100, probably not something to worry but could be investigated, but 1000 surely should be profiled carefully.